crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -838,16 +839,23 @@ def upgrade_controller_clusters
 
         return upgrade_controllers_disruptive if upgrade_mode == :normal
 
-        network_node = ::Node.find(
-          "pacemaker_founder:true AND " \
+        network_founders = ::Node.find(
+          "pacemaker_founder:* AND " \
           "run_list_map:neutron-network AND NOT " \
           "run_list_map:neutron-server"
-        ).first
+        )
+        network_node = ::Node.load(network_founders.first[:pacemaker][:founder]) rescue nil
         prepare_network_node(network_node) unless network_node.nil?
 
         # Now we must upgrade the clusters in the correct order:
         # 1. data, 2. API, 3. network
-        cluster_founders = ::Node.find("pacemaker_founder:true")
+        cluster_founders_names = ::Node.find("pacemaker_founder:*")
```

Comment:
"cluster_founders_names" actually means all nodes in all clusters, right?

Reply:
yes, but with the filtering below it turns into only founders names. I can use an extra var here or just chain all the methods below instead

Reply:
Or the search could be for something else, would `pacemaker:*` work as well?

Reply:
yes, but we want only the founders rigth? so we need to serach+filter here to extract the founders of all clusters

Reply:
My point is, this first search does not get you just founders anyway (you need to refine it later), so you do not need to look for "pacemaker_founder:*". 
But I guess it works, so it's just nitpicking

Reply:
no, but you need to search for all nodes in all clusters to get the founders names. We may be better with searching for `run_list_map:pacemaker-cluster-member` in here as well.